File: datasets_v2/tools/betterbatterystats/compose/text_widget_horz.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun text_widget_horz() {
  Column(modifier = Modifier.paint(painterResource(id = R.drawable.widget_shape), contentScale = ContentScale.Crop).fillMaxWidth().fillMaxHeight()) {
    Text(text = "StatType", modifier = Modifier.fillMaxWidth().wrapContentHeight(), fontSize = 10.sp, textAlign = TextAlign.Center)
    Row(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
      Row(modifier = Modifier.fillMaxWidth().fillMaxHeight(), verticalAlignment = Alignment.CenterVertically) {
        Row(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
          Column(modifier = Modifier.wrapContentHeight().weight(1.0f)) {
            Text(text = "Since", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "Deep Sleep", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), color = Color(0xFF2ECC71), fontSize = 10.sp)
            Text(text = "Screen on", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), color = Color(0xFFF1C40F), fontSize = 10.sp)
            Text(text = "Awake", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), color = Color(0xFFE74C3C), fontSize = 10.sp)
            Text(text = "KWL", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), color = Color(0xFFE67E22), fontSize = 10.sp)
            Text(text = "PWL", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), color = Color(0xFF54ACD2), fontSize = 10.sp)
          }
          Column(modifier = Modifier.wrapContentHeight().weight(1.0f)) {
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
            Text(text = "50m", modifier = Modifier.wrapContentWidth().wrapContentHeight().padding(start = 8.0.dp).padding(end = 0.0.dp), fontSize = 10.sp)
          }
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun text_widget_horz_preview() {
    MyApplicationTheme {
        text_widget_horz()
    }
}

File: experiments/tools/betterbatterystats/swiftui/Text_widget_horz.swift
import SwiftUI

struct Text_widget_horz: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      Text("StatType").font(.system(size: 10)).multilineTextAlignment(.center).frame(maxWidth: .infinity, alignment: .center)
      HStack(alignment: .center, spacing: 0) {
        HStack(alignment: .center, spacing: 0) {
          HStack(alignment: .center, spacing: 0) {
            VStack(alignment: .leading, spacing: 0) {
              Text("Since").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("Deep Sleep").font(.system(size: 10)).foregroundColor(Color(red: 0.1803921568627451, green: 0.8, blue: 0.44313725490196076, opacity: 1)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("Screen on").font(.system(size: 10)).foregroundColor(Color(red: 0.9450980392156862, green: 0.7686274509803922, blue: 0.058823529411764705, opacity: 1)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("Awake").font(.system(size: 10)).foregroundColor(Color(red: 0.9058823529411765, green: 0.2980392156862745, blue: 0.23529411764705882, opacity: 1)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("KWL").font(.system(size: 10)).foregroundColor(Color(red: 0.9019607843137255, green: 0.49411764705882355, blue: 0.13333333333333333, opacity: 1)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("PWL").font(.system(size: 10)).foregroundColor(Color(red: 0.32941176470588235, green: 0.6745098039215687, blue: 0.8235294117647058, opacity: 1)).padding(.leading, 8.0).padding(.trailing, 0.0)
            }
            .frame(maxWidth: .infinity)
            VStack(alignment: .leading, spacing: 0) {
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
              Text("50m").font(.system(size: 10)).padding(.leading, 8.0).padding(.trailing, 0.0)
            }
            .frame(maxWidth: .infinity)
          }
          .frame(maxWidth: .infinity)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
      }
      .frame(maxWidth: .infinity)
    }
    .background(Image("widget_shape").resizable().scaledToFill())
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Text_widget_horz_Previews: PreviewProvider {
  static var previews: some View {
    Text_widget_horz()
  }
}
